### PR TITLE
kubernetes: Add charts with node usage data

### PR DESCRIPTION
**Pull request by petervo**
Tooltips are missing but in the interest of getting this reviewed sooner I figured those can be a followup PR.

### CODE REVIEW

**Comment by andreasn** [comment]
Will need to work around this one somehow.

screenshot from 2016-05-16 15-43-55

**Comment by petervo** [comment]
@andreasn the charts in patternfly have the same issue. Do you have a suggestion of what you would like it to do?

**Comment by andreasn** [comment]
No, not sure yet. Is there a way we can decrease the font size if the string is long?
Another way to work around this exact one is to round the number off at 492 MiB instead of 492.1 MiB

File: pkg/kubernetes/styles/nodes.less [hunk 1]
```diff
@@ -13,3 +13,87 @@ div.nodes-delete ul {
...
+li.chart-legend-item span.legend-pf-text {
+    font-size: 11px;
+    font-weight: 400;
+    line-height: 11px;
+    margin-right: 5px;
+}
+
+svg {
+    min-width: 120px;
+    min-height: 120px;
```
Comment (stefwalter) [inline pkg/kubernetes/styles/nodes.less]: This breaks the topology view.

**Comment by stefwalter** [comment]
@andreasn With this change the multi-selection of nodes is very far away from the nodes themselves. It seems awkward. Is that okay?

@petervo The view with only one node is rather verbose and confusing. Is there a way we should react to the number of nodes and minimize the size of the top left hand Memory/CPU/Disk heatmap?

**Comment by petervo** [comment]
> @petervo <https://github.com/petervo> The view with only one node is
> rather verbose and confusing. Is there a way we should react to the
> number of nodes and minimize the size of the top left hand
> Memory/CPU/Disk heatmap?

What would you recommend? Keep it at 50% until there are more than 5 nodes?

**Comment by andreasn** [comment]
This area looks a lot taller than the mockups.

screenshot from 2016-05-17 12-36-57

**Comment by andreasn** [comment]
I would suggest making the squares 32x32 instead of 64x64

**Comment by petervo** [comment]
The square adjust size when there are more do you want 32x32 to be max size?

**Comment by petervo** [comment]
Changed the height to 90. Now the label on the pie looks terrible.

**Comment by andreasn** [comment]
> The square adjust size when there are more do you want 32x32 to be max size?

Ah, no, that makes sense.

**Comment by andreasn** [comment]
I played around with overriding some margins and paddings and came up with this.

screenshot from 2016-05-17 13-22-20

**Comment by petervo** [comment]
The small circle is what makes things look weird, but i thought you wanted the heat map and the donut to have the same height. Anyways if you send me the css you want I can put that in instead but there's still the font size issue for the usage donuts so that might need to be fixed anyways

**Comment by andreasn** [comment]
Sound good. I'll get something together asap.

**Comment by petervo** [comment]
Added some code to automatically resize the center label

File: pkg/kubernetes/scripts/charts.js [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+                            });
+
+                        newLi.append("span")
+                            .classed('legend-pf-color-box', true);
+
+                        newLi.append("span")
+                            .classed('legend-pf-text', true);
+
+                        li.attr("data-color", function (d, i) {
+                                return colors[i];
```
Comment (stefwalter) [inline pkg/kubernetes/scripts/charts.js]: Indentation is broken here.

File: pkg/kubernetes/scripts/charts.js [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+                    var pie = d3.layout.pie().value(function (d) {
+                        if (typeof d === 'object')
+                            return d.value;
+                        else
+                            return d;
+                    });
+
+                    var legend;
+                    if (attributes['legend']) {
+                        legend = d3.select("#"+attributes['legend'])
```
Comment (stefwalter) [inline pkg/kubernetes/scripts/charts.js]: Fixing spacing here.

File: pkg/kubernetes/scripts/charts.js [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+                            title.insert('tspan').text($scope.smallTitle)
+                                .classed('donut-title-small-pf', true)
+                                .attr('dy', 20).attr('x', 0)
+                                .style("font-size", calFontSize);
+                        }
+                    }
+
+                    function select(id) {
+                        var sel = "path[data-id='"+ id+ "']";
+                        var lsel = "li[data-id='"+ id+ "']";
```
Comment (stefwalter) [inline pkg/kubernetes/scripts/charts.js]: Fixing indentation.

File: pkg/kubernetes/scripts/charts.js [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+                        }
+                    }
+
+                    function unselect() {
+                       g.selectAll("path")
+                          .classed(focusResetClasses)
+                          .attr("d", arc);
+                        if (legend)
+                            legend.selectAll("li")
+                                .classed(focusResetClasses);
```
Comment (stefwalter) [inline pkg/kubernetes/scripts/charts.js]: Fixing indentation.

**Comment by stefwalter** [comment]
Merged.

**Pull request closed without merging**